mabwiser/utils.py

```diff
@@ -60,3 +62,148 @@ def reset(dictionary: Dict, value) -> NoReturn:
+                from the distribution (see above for behavior if ``high=None``).
+                If array-like, must contain integer values
+            size : int or tuple of ints, optional
+                Output shape.  If the given shape is, e.g., ``(m, n, k)``, then
+                ``m * n * k`` samples are drawn.  Default is None, in which case a
+                single value is returned.
+
+            Returns
+            -------
+            out : int or ndarray of ints
+                `size`-shaped array of random integers from the appropriate
+                distribution, or a single such random int if `size` not provided.
+        """
+        pass
+
+    @abc.abstractmethod
+    def choice(self, a: Union[int, Iterable[int]], size: Union[int, Tuple[int]] = None, p: Iterable[float] = None):
+        """ Return a random sample from a given 1-D array
+            Based on the probabilities associated with each entry in the array
+
+            Parameters
+            ----------
+            a : 1-D array-like or int
+                If an ndarray, a random sample is generated from its elements.
+                If an int, the random sample is generated as if a were np.arange(a)
+            size : int or tuple of ints, optional
+                    Output shape.  If the given shape is, e.g., ``(m, n, k)``, then
+                    ``m * n * k`` samples are drawn.  Default is None, in which case a
+                    single value is returned.
+            p : 1-D array-like, optional
+                The probabilities associated with each entry in options.
+
+            Returns
+            -------
+            samples : single item or ndarray
+                The generated random samples
+        """
+
+        pass
+
+    @abc.abstractmethod
+    def beta(self, alpha: int, beta: int):
+        """ Return a sample from a Beta distribution.
+
+            Parameters
+            ----------
+            alpha : float or array_like of floats
+                    Alpha, positive (>0).
+            beta : float or array_like of floats
+                    Beta, positive (>0).
+
+            Returns
+            -------
+            out : scalar
+                Drawn sample from the parameterized beta distribution.
+        """
+        pass
+
+    @abc.abstractmethod
+    def standard_normal(self, size):
+        """ Draw samples from a standard Normal distribution (mean=0, stdev=1).
+
+            Parameters
+            ----------
+            size : int or tuple of ints
+                Output shape.  If the given shape is, e.g., ``(m, n, k)``, then
+                ``m * n * k`` samples are drawn.
+
+            Returns
+            -------
+            out : ndarray
+                A floating-point array of shape ``size`` of drawn samples
+        """
+        pass
+
+
+class _NumpyRNG(_BaseRNG):
+
+    def __init__(self, seed):
+        super().__init__(seed)
+        self.rng = np.random.RandomState(self.seed)
```

Comment:
Think we'll need to try this with the new Numpy [Random Generator](https://numpy.org/doc/1.18/reference/random/generator.html) class to see if it works.

Reply:
Agreed. 

But may I define the scope of this PR narrower than that: replicate the current functionality as-is. Uptaking a new rng sounds like next update to me. 

That said if you two claim that this PR would not add much without also upgrading the RNG I can pull back. 

Thoughts @erstrong @bkleyn

Reply:
My understanding is that the goal for this was to create a wrapper class that would allow us to swap out random generators without needing to change other code. The best way to do that is to test it with the Generator class, since there is at least one major difference that I am aware of (see my comment about seeds) that would require a function in the wrapper class, and there may be others.

Reply:
My concern was the uptaking the rng would not be trivial and that additional functionality would be needed to deal random states for parallelization. However, after a closer look I think it's simpler than we previously thought. So I'm good with merging this PR and then uptaking the new rng as a separate PR.